tests/test_slots.py

```diff
@@ -0,0 +1,26 @@
+from falcon import Request, Response
+import falcon.testing as testing
+
+
+class TestSlots(testing.TestBase):
+
+    def __init__(self):
+        self.failed = False
+        super(TestSlots, self).__init__()
+
+    def test_slots_request(self):
+        env = testing.create_environ()
+        req = Request(env)
+        try:
+            req.doesnt = 'exist'
+        except AttributeError:
+            self.failed = True
```

Comment:
My thought was that you could remove the `self.failed` variable altogether:

```py
def test_slots_response(self):
        resp = Response()
        try:
            resp.doesnt = 'exist'
        except AttributeError:
            self.fail('Unable to add to __slots__ dynamically')
```

Even better (now that I think of it) would be to use [ExpectedException](http://testtools.readthedocs.io/en/latest/for-test-authors.html#expectedexception) (both `falcon.testing.TestBase` and `falcon.testing.TestCase` inherit from `testtools.TestCase).